coreneuron/nrniv/nrn_setup.cpp

```diff
@@ -1137,9 +1140,20 @@ void read_phase2(FileHandler& F, int imult, NrnThread& nt) {
         nmech = F.read_int();
         tml_index = new int[nmech];
         ml_nodecount = new int[nmech];
+        int diff_modfiles_count = 0;
         for (int i = 0; i < nmech; ++i) {
             tml_index[i] = F.read_int();
             ml_nodecount[i] = F.read_int();
+            auto found_different_mod_file =
+                std::find(different_mod_files.begin(), different_mod_files.end(), tml_index[i]);
+            if (found_different_mod_file != different_mod_files.end()) {
```

Comment:
I have added the `std::find() != different_mechanism_type.end()` expression inside the if and I hope it is still understandable.
I have also got a question regarding this check only by rank 0. If `read_phase2()` is called by all threads, shouldn't all the ranks calculate the number of different files, so that they can all call `MPI_Finalize()` if there is a different mechanism and then `exit()`? 
Please correct me if I am wrong, but from a bit of study I found out that this is the case and added a `graceful_exit()` function that all ranks call if they need to exit. 
I have corrected the other parts of the code and only rank 0 prints to stdout.